src/components/pip/pip.tsx

```diff
@@ -19,36 +21,66 @@ interface PIPComponentOwnProps {
   hide: () => void;
   onSideBySideSwitch?: () => void;
   onInversePIP?: () => void;
+  dragAndSnapManager?: DragAndSnapManager;
 }
 interface PIPComponentConnectProps {
-  guiClientRect?: {height: number; width: number};
+  guiClientRect?: GuiClientRect;
   playerHover?: boolean;
 }
+interface PIPComponentState {
+  isDragging: boolean;
+}
 type PIPComponentProps = PIPComponentOwnProps & PIPComponentConnectProps;
 @connect(mapStateToProps, utils.bindActions(reducers.shell.actions))
-export class Pip extends Component<PIPComponentProps> {
-  ref = createRef();
+export class Pip extends Component<PIPComponentProps, PIPComponentState> {
+  videoContainerRef = createRef<HTMLDivElement>();
+  pipContainerRef = createRef<HTMLDivElement>();
+  constructor(props: PIPComponentProps) {
+    super(props);
+    this.state = {
+      isDragging: false
+    };
+  }
 
   componentDidMount() {
-    this.ref.current.appendChild(this.props.childPlayer.getView());
+    const {childPlayer, dragAndSnapManager} = this.props;
+    this.videoContainerRef?.current?.appendChild(childPlayer.getView());
+    if (dragAndSnapManager && this.pipContainerRef.current) {
```

Comment:
Do we need this check? dragAndSnapManager is not always not null?

Reply:
of course, for Main PIP we don't need it